lib/rex/parser/nmap_nokogiri.rb

```diff
@@ -1,404 +1,472 @@
-              :workspace => wspace,
-              :host => host_object,
-              :service => svc_obj,
-              :type => "nmap.nse.#{k}." + (svc[:proto] || "tcp") +".#{svc[:port]}",
-              :data => { 'output' => v },
-              :update => :unique_data
-            }
-            db_report(:note, nse_note)
-          end
-        end
-        reported << svc_obj
-      end
-      reported
-    end
+                       include NokogiriDocMixin
+
+                       def determine_port_state(v)
+                         case v
+                         when 'open'
+                           Msf::ServiceState::Open
+                         when 'closed'
+                           Msf::ServiceState::Closed
+                         when 'filtered'
+                           Msf::ServiceState::Filtered
+                         else
+                           Msf::ServiceState::Unknown
+                         end
+                       end
+
+                       # Compare OS fingerprinting data
+                       def better_os_match(orig_hash, new_hash)
+                         return false unless new_hash.key? 'accuracy'
+                         return true unless orig_hash.key? 'accuracy'
+
+                         new_hash['accuracy'].to_i > orig_hash['accuracy'].to_i
+                       end
+
+                       # Triggered every time a new element is encountered. We keep state
+                       # ourselves with the @state variable, turning things on when we
+                       # get here (and turning things off when we exit in end_element()).
+                       def start_element(name = nil, attrs = [])
+                         attrs = normalize_attrs(attrs)
+                         block = @block
+                         @state[:current_tag][name] = true
+                         case name
+                         when 'status'
+                           record_host_status(attrs)
+                         when 'address'
+                           record_address(attrs)
+                         when 'osclass'
+                           record_host_osclass(attrs)
+                         when 'osmatch'
+                           record_host_osmatch(attrs)
+                         when 'uptime'
+                           record_host_uptime(attrs)
+                         when 'hostname'
+                           record_hostname(attrs)
+                         when 'port'
+                           record_port(attrs)
+                         when 'state'
+                           record_port_state(attrs)
+                         when 'service'
+                           record_port_service(attrs)
+                         when 'script'
+                           record_port_script(attrs)
+                           record_host_script(attrs)
+                           record_vuln_script(attrs)
+                           # Ignoring post scripts completely
```

Comment:
I'm marking this as unresolved purely because I want to know if your introduction of `record_vuln_script` invalidates this comment or not. If this is not the case then feel free to mark this comment as resolved and move on. This is more of just a double check/flagging a potential issue on my part.

Reply:
Closing this issue as we don't know what this comment was left in here for and we aren't comfortable removing it.